components/datetime/src/time_zone.rs

```diff
@@ -251,59 +387,246 @@ impl TimeZoneFormat {
+                SpecificNonLocationShortFormat {},
+            ));
+        Ok(self)
+    }
+
+    /// Load generic location format for timezone. For example, Los Angeles Time.
+    pub fn load_generic_location_format<ZP>(
+        &mut self,
+        zone_provider: &ZP,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError>
+    where
+        ZP: ResourceProvider<provider::time_zones::ExemplarCitiesV1Marker> + ?Sized,
+    {
+        if self.data_payloads.exemplar_cities == None {
+            load_resource(
+                &self.locale,
+                &mut self.data_payloads.exemplar_cities,
+                zone_provider,
+            )?;
+        }
+        self.format_units.push(TimeZoneFormatUnit::GenericLocation(
+            GenericLocationFormat {},
+        ));
+        Ok(self)
+    }
+
+    /// Load exemplar city format for timezone. For example, Los Angeles.
+    fn load_exemplar_city_format<ZP>(
+        &mut self,
+        zone_provider: &ZP,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError>
+    where
+        ZP: ResourceProvider<provider::time_zones::ExemplarCitiesV1Marker> + ?Sized,
+    {
+        if self.data_payloads.exemplar_cities == None {
+            load_resource(
+                &self.locale,
+                &mut self.data_payloads.exemplar_cities,
+                zone_provider,
+            )?;
+        }
+        self.format_units
+            .push(TimeZoneFormatUnit::ExemplarCity(ExemplarCityFormat {}));
+        Ok(self)
+    }
+
+    /// Load localized GMT format for timezone. For example, GMT-07:00.
+    pub fn load_localized_gmt_format(
+        &mut self,
+        is_fallback: bool,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError> {
+        if is_fallback {
+            self.fallback_unit = Some(TimeZoneFormatUnit::LocalizedGmt(LocalizedGmtFormat {}));
+        } else {
+            self.format_units
+                .push(TimeZoneFormatUnit::LocalizedGmt(LocalizedGmtFormat {}));
+        }
+        Ok(self)
+    }
+
+    /// Load Iso8601 format for timezone. For example, -07:00.
+    pub fn load_iso_8601_format(
+        &mut self,
+        format: IsoFormat,
+        minutes: IsoMinutes,
+        seconds: IsoSeconds,
+        is_fallback: bool,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError> {
+        if is_fallback {
+            self.fallback_unit = Some(TimeZoneFormatUnit::Iso8601(Iso8601Format {
+                format,
+                minutes,
+                seconds,
+            }));
+        } else {
+            self.format_units
+                .push(TimeZoneFormatUnit::Iso8601(Iso8601Format {
+                    format,
+                    minutes,
+                    seconds,
+                }));
+        }
+        Ok(self)
+    }
+
+    /// Load a fallback format for timezone. The fallback format will be executed if there are no
+    /// matching format results.
+    pub fn load_fallback_format(
```

Comment:
> thought

Hmm, this function itself is convenient, but also unnecessary since the `gmt` and `iso` functions take a parameter regarding whether its for fallback or not. 

I think I'd rather see the boolean option removed from the `load_gmt` and `load_iso` functions and have this function just inline the code to add it to the fallback.

That way we don't have to deal with the boolean/enum argument on those functions, and the code to handle this is not very big at all.